server/src/main/java/org/opensearch/index/store/lockmanager/RemoteStoreMetadataLockManager.java

```diff
@@ -131,4 +132,19 @@ public void cloneLock(LockInfo originalLockInfo, LockInfo clonedLockInfo) throws
     public void delete() throws IOException {
         lockDirectory.delete();
     }
+
+    private String[] getLockFiles(String[] lockDirectoryContents) throws IOException {
+        if (lockDirectoryContents == null || lockDirectoryContents.length == 0) {
+            return new String[0];
+        }
+        // filtering lock files from lock directory contents.
+        // this is a good to have check, there is no known prod scenarios where this can happen
```

Comment:
nit : lets not use `prod` word  . Also a user can create extra files in the lock folder as well